### PR TITLE
PJR-137 Setup R2 dependency

### PR DESCRIPTION
**Pull request by ettore**
This makes the app compile R2 and link it to the final executable. 

I added R2 as a remote SPM package for now, since it's going to be a while until we actually need to modify that codebase. If we need to, we can change this to a git submodule. 

This also adds an empty "NYPLReader" SPM package. This will be the actual eReader package. Currently it only includes references to the R2 packages to verify linking works fine. I verified R2 is being built and NYPLReader's unit tests are being run.

### CODE REVIEW

File: apps/oeh/ios/App/Podfile [hunk 1]
```diff
@@ -15,13 +15,16 @@ def capacitor_pods
   pod 'CapacitorFilesystem', :path => '../../../../node_modules/@capacitor/filesystem'
   pod 'CapacitorPreferences', :path => '../../node_modules/@capacitor/preferences'
   pod 'CapacitorSplashScreen', :path => '../../../../node_modules/@capacitor/splash-screen'
+  pod 'CordovaPlugins', :path => '../capacitor-cordova-ios-plugins'
```
Comment (ettore) [inline apps/oeh/ios/App/Podfile:18]: I think I had this removed in the past, but every time I run `npm run build`/`npm run sync:static` this is being added back, so I figure it should stay.

File: apps/oeh/ios/App/App.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -402,7 +407,7 @@
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SDKROOT = iphoneos;
 				SWIFT_COMPILATION_MODE = wholemodule;
-				SWIFT_OPTIMIZATION_LEVEL = "-O";
+				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
```
Comment (ettore) [inline apps/oeh/ios/App/App.xcodeproj/project.pbxproj:410]: this is an Xcode recommended setting for release builds.

File: apps/oeh/ios/App/Podfile (hunk 1 above)
Comment (kristojorg) [reply apps/oeh/ios/App/Podfile:18]: I finally removed the npm package I had installed that is triggering it to be added back. Not sure if we will eventually need it or not, but for now it's safe to ignore

File: apps/oeh/ios/App/Podfile [hunk 3]
```diff
@@ -15,13 +15,16 @@ def capacitor_pods
...
   capacitor_pods
   # Add your Pods here
 end
 
-# post_install do |installer|
-#   assertDeploymentTarget(installer)
-# end
+post_install do |installer|
+  installer.pods_project.build_configurations.each do |config|
+    config.build_settings['CLANG_WARN_DOCUMENTATION_COMMENTS'] = ['NO']
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply apps/oeh/ios/App/Podfile:28]: 👍 gratzie

**Review by kristojorg — APPROVED** [review]

File: apps/oeh/ios/App/Podfile (hunk 1 above)
Comment (ettore) [reply apps/oeh/ios/App/Podfile:18]: sounds good -- I think what we'll do is update the podfile every time npm changes it, so that works.

**Pull request merged**